src/foundry/client/data/documents/macro.d.mts

```diff
@@ -34,32 +51,24 @@ declare global {
      */
     get thumbnail(): string | null;
 
+    /**
+     * Test whether the given User is capable of executing this Macro.
+     * @param user - The User to test.
+     * @returns Can this User execute this Macro?
+     */
+    canUserExecute(user: User): boolean;
+
     /**
      * Execute the Macro command.
      * @param scope - Macro execution scope which is passed to script macros
      * @returns A promising containing a created {@link ChatMessage} (or `undefined`) if a chat
      *          macro or the return value if a script macro. A void return is possible if the user
      *          is not permitted to execute macros or a script macro execution fails.
      */
-    execute(scope?: Scope): Promise<ChatMessage | undefined> | Promise<unknown> | void;
+    execute(scope?: InexactPartial<Macro.Scope>): Promise<ChatMessage | undefined> | Promise<unknown> | void;
```

Comment:
This predates your changes but do you happen to know when it returns something besides `ChatMessage | undefined` in the promise? Is it like the return result of a user's macro...?